### ISSUE TITLE
aabb's ray_intersect always returning gid -1, id 0

### ISSUE THREAD

**Issue by luca-penasa** [libigl/libigl#339]
I've been able to successfully use  libigl to test triangle to ray intersection in this way:
```cpp
ray_mesh_intersect(camera->getCenter(), -tri_to_cam, verts, faces, hits);
```
but I was trying to move to an AABB, here I am initializing the aabb structure:
```cpp
igl::AABB<MatrixXd,3> * tree = new igl::AABB<MatrixXd,3>();
	tree->init(mesh->getVertices(), mesh->getFaces());
```

and then in the code I use it instead of the previous ray_mesh_intersect call in this way:
```cpp
Eigen::Matrix<double, -1, 3> origin = camera->getCenter().transpose();
			tree->intersect_ray(verts, faces, origin, -tri_to_cam , hits);
```

this code is returning a variable number of hits (which might be plausible) but gid and id are set to -1 and 0 for all the hists. Is this piece of code used somewhere else and know to be working? Could anybody have a fast check if the code is working for them?

Many thanks
Luca

**Comment by luca-penasa** [comment]
I solved using igl::embree::EmbreeIntersector
But maybe is a good idea to havea look at the pure igl algo

**Comment by alecjacobson** [comment]
Thanks. This was indeed a bug. It's been fixed in 367220022fcc55858601e6036ee413e8baec24ea
